### PR TITLE
RavenDB-6707 Add hash to the File Header

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

File: src/Voron/Impl/Backup/VoronBackupUtil.cs [hunk 1]
```diff
@@ -17,6 +17,10 @@ internal static void CopyHeaders(CompressionLevel compression, ZipArchive packag
                 if (!storageEnvironmentOptions.ReadHeader(headerFileName, header))
                     continue;
 
+                var hash = HeaderAccessor.CalculateFileHeaderHash(header);
```
Comment (ayende) [inline src/Voron/Impl/Backup/VoronBackupUtil.cs]: No, this should be moved to the `ReadHeader` method.
The exception should be moved for the bottom of this method, if we failed to read both header files

File: src/Voron/Impl/FileHeaders/FileHeader.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ public struct FileHeader
         /// to be written to a single sector, because we assume atomic sector writes.
         /// </summary>
         private static readonly unsafe byte[] AssertTransactionHeaderSize = new byte[sizeof(FileHeader) < 512 ? 0 : -1];
+        public static int HashOffset = (int)Marshal.OffsetOf<FileHeader>(nameof(Hash));
```
Comment (ayende) [inline src/Voron/Impl/FileHeaders/FileHeader.cs:17]: Add assert that it is always the last thing on the file header

File: src/Voron/Impl/FileHeaders/HeaderAccessor.cs [hunk 3]
```diff
@@ -75,7 +77,11 @@ public bool Initialize()
                 {
                     *f2 = *f1;
                 }
-
+                
+                var hash = CalculateFileHeaderHash(f1);
+                if (f1->Hash != hash)
```
Comment (ayende) [inline src/Voron/Impl/FileHeaders/HeaderAccessor.cs:82]: No, we should also check if we are valid of `f2` and use that

**Review by ayende — CHANGES_REQUESTED** [review]

**Comment by iftahbe** [comment]
Many tests are failing, I will close, merge and reopen the PR

**Pull request closed without merging**